spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpNioConnection.java

```diff
@@ -620,11 +625,27 @@ public void write(int b) throws IOException {
 
 		@Override
 		public void close() {
-			doClose();
+			if (this.selector != null) {
+				try {
+					this.selector.close();
+				}
+				catch (@SuppressWarnings(UNUSED) Exception e) {
+				}
+			}
+			try {
+				super.close();
```

Comment:
No reason to call `super`: it is just an empty method.
That's why we didn't delegate to it originally anyway.